index.js

```diff
@@ -98,6 +99,27 @@ app.post("/github_discord_urls", async (req, res) => {
   }
 });
 
+async function insert_user_details_to_db() {
+  const data_4 = await pool.query(
+    `insert into auth_user_info (username, password) values ($1, $2)`,
+    [db_username, db_password]
+  );
+}
+
+app.post("/user_signup", async (req, res) => {
+  try {
+    const reqBody = req.body;
+    const hashedPassword = await bcrypt.hash(reqBody.password, 10);
+    await insert_user_details_to_db(
```

Comment:
have to check if this user email already exists in the db; if yes, you have to reject

Reply:
It would be rejected because of the unique property in db

Reply:
right - but that would throw a db exception, which will be a 500. Ideally, the server should catch it and throw a 400

Reply:
ok